src/components/ImageItem/ImageItem.ios.tsx

```diff
@@ -61,18 +70,24 @@ const ImageItem = ({
   );
   const imageStylesWithOpacity = { ...imagesStyles, opacity: imageOpacity };
 
-  const onScrollEndDrag = ({
-    nativeEvent
-  }: NativeSyntheticEvent<NativeScrollEvent>) => {
-    const velocityY = nativeEvent?.velocity?.y ?? 0;
-    const isZoomed = nativeEvent?.zoomScale > 1;
-
-    onZoom(isZoomed);
-
-    if (!isZoomed && Math.abs(velocityY) > SWIPE_CLOSE_VELOCITY) {
-      onRequestClose();
-    }
-  };
+  const onScrollEndDrag = useCallback(
+    ({ nativeEvent }: NativeSyntheticEvent<NativeScrollEvent>) => {
+      const velocityY = nativeEvent?.velocity?.y ?? 0;
+      const scaled = nativeEvent?.zoomScale > 1;
+
+      onZoom(scaled);
```

Comment:
Seem to better use useEffect to avoid constant callback call on every drag.

Reply:
Actually you can't use `useEffect` inside `useCallback` hook. So I better leave as it is, anyway parent component is using `useState` that prevent from rerenders if state has not changed

Reply:
I didn't propose you to put `useEffect` inside `useCallback`. :)
I meant to use it as all hooks.
`const [scaled, setScaled] = useState(false);
useEffect(() => onZoom(scaled), [scaled, onZoom])`